res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,846 @@
+const Prime4ColorMapper = new ColorMapper({
+    0x000020: 0x01, // dark blue
+    0x0000AA: 0x02, // dim blue
+    0x0000C0: 0x03, // blue
+    0x002000: 0x04, // dark green
+    0x002020: 0x05, // dark cyan
+    0x0020A0: 0x06,
+    0x0020C0: 0x07,
+    0x00A000: 0x08, // dim green
+    0x00A020: 0x09,
+    0x00A0A0: 0x0A, // dim cyan
+    0x00A0C0: 0x0B,
+    0x00C000: 0x0C, // green
+    0x00C020: 0x0D,
+    0x00C0A0: 0x0E,
+    0x00C0C0: 0x0F, // cyan
+    0x200000: 0x10, // dark red
+    0x200020: 0x11, // dark magenta
+    0x2000A0: 0x12,
+    0x2000C0: 0x13,
+    0x2020A0: 0x16,
+    0x2020C0: 0x17,
+    0x20A000: 0x18,
+    0x20A020: 0x19,
+    0x20A0A0: 0x1A,
+    0x20A0C0: 0x1B,
+    0x20C000: 0x1C,
+    0x20C020: 0x1D,
+    0x20C0A0: 0x1E,
+    0x20C0C0: 0x1F,
+    0xA00000: 0x20, // dim red
+    0xA00020: 0x21,
+    0xA000A0: 0x22, // dim magenta
+    0xA000C0: 0x23,
+    0xA02000: 0x24,
+    0xA02020: 0x25,
+    0xA020A0: 0x26,
+    0xA020C0: 0x27,
+    0xA0A000: 0x28, // dim yellow
+    0xA0A020: 0x29,
+    0xA0A0A0: 0x2A, // dim white/grey
+    0xA0A0C0: 0x2B,
+    0xA0C000: 0x2C,
+    0xA0C020: 0x2D,
+    0xA0C0A0: 0x2E,
+    0xA0C0C0: 0x2F,
+    0xC00000: 0x30, // red
+    0xC00020: 0x31,
+    0xC000A0: 0x32,
+    0xC000C0: 0x33, // purple
+    0xC02000: 0x34, // orange
+    0xC02020: 0x35,
+    0xC020A0: 0x36,
+    0xC020C0: 0x37,
+    0xC0A000: 0x38,
+    0xC0A020: 0x39,
+    0xC0A0A0: 0x3A, // pink
+    0xC0A0C0: 0x3B,
+    0xC0C000: 0x3C, // yellow
+    0xC0C020: 0x3D,
+    0xC0C0A0: 0x3E,
+    0xC0C0C0: 0x3F, // white
+});
+
+// Set active + inactive values for user-defined deck colours
+const colDeck = [
+    Prime4.rgbCode[deckColors[0]],
+    Prime4.rgbCode[deckColors[1]],
+    Prime4.rgbCode[deckColors[2]],
+    Prime4.rgbCode[deckColors[3]],
+];
+const colDeckDark = [
+    Prime4.rgbCode[deckColors[0] + "Dark"],
+    Prime4.rgbCode[deckColors[1] + "Dark"],
+    Prime4.rgbCode[deckColors[2] + "Dark"],
+    Prime4.rgbCode[deckColors[3] + "Dark"],
+];
+
+// Register '0x9n' as a button press and 'ox8n' as a button release
+components.Button.prototype.isPress = function(channel, control, value, status) {
+    return (status & 0xF0) === 0x90;
+};
+
+Prime4.DeckAssignButton = function(options) {
+    components.Button.call(this, options);
+
+    if (!Number.isInteger(this.deckIndex)) {
+        throw `invalid deckIndex: ${this.deckIndex}`;
+    }
+    if (!(this.toDeck instanceof Prime4.Deck)) {
+        throw "invalid toDeck";
+    }
+
+    const deckSide = (this.deckIndex % 2) === 0 ? "leftDeck" : "rightDeck";
+    if (!(Prime4[deckSide] instanceof Prime4.Deck)) {
+        throw "invalid deckIndex or structure; We expect Prime4.leftDeck and Prime4.rightDeck to be valid decks representing the physical left and right decks";
+    }
+    const isActive = function() {
+        return Prime4[deckSide] === this.toDeck;
+    };
+
+    this.trigger = function() {
+        this.output(isActive());
+    };
+
```

Comment:
Please add this to help debug the problem that the leds stay dim.
```suggestion

    this.outValueScale = function (value) {
         print(value);
         print(deckSide);
         const scaledVal = components.Button.prototype.outValueScale.call(this, value);
         print(scaledVal);
         return scaledVal;
    }
```